src/index.ts

```diff
@@ -12,7 +12,7 @@ const fetchWithRetry = (url: string, init: RequestInit, maxRetries: number = 3):
           .catch(() => ({ response, error: { message: 'Unable to parse JSON' } }))
     }).then(({ error }: any) => {
       if (error) {
-        throw Error(error)
+        throw Error(error.message)
```

Comment:
Thanks for contributing. This fix was necessary.

However, wouldn't it be better if we also make a fallback ?
```suggestion
        throw new Error(error.message || 'Unknown error')
```